File: 7_kyu/Dont_give_me_five.rs
fn dont_give_me_five(start: isize, end: isize) -> isize {
    (start..=end).filter(no_digit_5).count() as isize
}

fn no_digit_5(x: &isize) -> bool {
    let mut remain = x.abs();
    while remain > 0 {
        if remain % 10 == 5 {
            return false;
        }
        remain /= 10;
    }
    true
}

File: 7_kyu/Dont_give_me_five.go
package kata

// Has the decimal number a digit?
func hasDigit(number, digit int) bool {
	if number < 0 {
		number = -number
	}
	for number > 0 {
		if number%10 == digit {
			return true
		}
		number /= 10
	}
	return false
}

// Counts all decimal numbers in range start ... end which have no digit 5.
func DontGiveMeFive(start int, end int) int {
	count := 0
	for i := start; i <= end; i++ {
		if !hasDigit(i, 5) {
			count++
		}
	}
	return count
}
